torch_geometric/nn/conv/gat_conv.py

```diff
@@ -176,10 +180,22 @@ def __init__(
             self.lin_edge = None
             self.register_parameter('att_edge', None)
 
-        if bias and concat:
-            self.bias = Parameter(torch.empty(heads * out_channels))
-        elif bias and not concat:
-            self.bias = Parameter(torch.empty(out_channels))
+        # The number of output channels:
+        total_out_channels = out_channels * (heads if concat else 1)
+
+        if residual:
+            self.res = Linear(
+                in_channels
+                if isinstance(in_channels, int) else in_channels[1],
+                total_out_channels,
+                bias=False,
+                weight_initializer='glorot',
+            )
+        else:
+            self.register_parameter('res', None)
```

Comment:
In this way, a linear transformation will also be applied when `in_channels == total_out_channels`. It is consistent with the original `example/ppi.py` but over-complicates the idea of skip-connection.

Maybe it's better to directly add the residual without linear transformation following the GAT authors' official implementation  here:
> https://github.com/PetarV-/GAT/blob/5af87e7fce2b90ae1cbd621cd58059036a3c7436/utils/layers.py#L27-L32